braindecode/models/dilated_conv_decoder.py

```diff
@@ -0,0 +1,1097 @@
+from functools import partial
+
+import torch
+import torchaudio as ta
+from torch import nn
+from torch.nn import functional as F
+
+from braindecode.models.base import EEGModuleMixin
+from braindecode.modules.attention import LocalSelfAttention
+from braindecode.modules.layers import SubjectLayers
+
+__all__ = ["DilatedConvDecoder"]
+
+
+class DilatedConvDecoder(EEGModuleMixin, nn.Module):
+    r"""Dilated Convolutional Decoder (SimpleConv) as the Brain Module from [brainmagik]_.
+
+    :bdg-secondary:`Recurrent` :bdg-info:`Small Attention` :bdg-success:`Convolution`
+
+    .. figure:: ../_static/model/simpleconv.png
+        :align: center
+        :alt: SimpleConv Architecture
+        :width: 1000px
+
+        For each layer, we note first the number of output channels, while the number of time steps is constant throughout the layers.
+        The model is composed of a spatial attention layer, then a 1x1 convolution without activation.
+        A 'Subject Layer' is selected based on the subject index s, which consists in a 1x1 convolution learnt only for that subject with no activation.
+        Then, we apply five convolutional blocks made of three convolutions.
+        The first two use residual skip connection and increasing dilation, followed by a BatchNorm layer and a GELU activation.
+        The third convolution is not residual, and uses a GLU activation (which halves the number of channels) and no normalization.
+        Finally, we apply two 1x1 convolutions with a GELU in between.
+
+    The **Dilated Convolutional Decoder**, sometimes referred to as SimpleConv [brainmagik]_,
+    functions as the **Brain Module** (or deep convolutional network) described in the
+    original paper by Défossez et al. (2023) for decoding speech perception from
+    non-invasive MEG/EEG recordings.
+
+    The model is a flexible encoder-decoder architecture adapted for Braindecode's
+    EEG workflows, originally designed for MEG decoding [brainmagik]_. Its primary role is to
+    map raw brain activity :math:`X \in \mathbb{R}^{C \times T}` into a latent
+    representation :math:`Z \in \mathbb{R}^{F \times T}` that is maximally aligned
+    with deep speech representations (e.g., wav2vec 2.0 features) using a contrastive
+    learning objective (CLIP loss).
+
+    The architecture leverages **dilated convolutions** to achieve large receptive
+    fields while maintaining temporal resolution [brainmagik]_.
+
+    Architectural Overview
+    ----------------------
+
+    The Brain Module processes input through a series of stages: input conditioning,
+    encoding via dilated convolution blocks, optional temporal processing (LSTM and
+    attention), and a symmetrical decoder.
+
+    1. **Input Conditioning:** Handles Spectrogram transformation (optional STFT/n_fft),
+       Channel Dropout, Subject-Specific Embeddings, and initial 1x1 convolutions.
+    2. **Encoder:** A series of dilated convolutional blocks that downsample the signal.
```

Comment:
Since the blocks are residual the time dimension doesn't shrink, so no effective downsampling.